### ISSUE TITLE
How to compile Nemo

### ISSUE THREAD

**Issue by wyattbiker** [linuxmint/nemo#2064]
Hello, I am using LM 19.1 with Cinnamon. What are the instructions for compiling Nemo? Thanks!

**Comment by brownsr** [comment]
take a look at mint-dev-tools, it's a repository here, you can install it through the software manager.  mint-dev-setup will install the main dependencies, mint-dev-build gives various build options.  If you are happy working it out more directly then clone nemo, dpkg-buildpackage within the repository directory, and the deb will go to the directory above for you to install.  If you find missing dependencies in the build you will need to install those along the way, usually just sudo apt-get install [thingy], build-essential is a package that you will definitely need as without that you won't get past square one.   Watch out that it's all set up to build off master, so you will find cross repository dependencies as development continues from the release version.  If you are building on another distribution then things may get more complicated, but I assume you are on the main mint distro from what you said.  If you muck things up and break the UI make sure you know how to get to a command line login so you can use terminal commands, so you can run mintupdate from the command line, that's generally an easy way to get back to the main current packages to reinstall, reboot, and recover.  Usually something like alt-ctrl-f1 or f2, and then alt-ctrl-f7 to get back to the current session, may vary according to your setup

Hope this helps

Regards

Simon

**Comment by wyattbiker** [comment]
Thank you for the advise!

**Comment by geajack** [comment]
I tried running `dpkg-buildpackage` in the root of this repository, and it dumped a bunch of files in the parent directory, but also gave me the error:

```
gpg: skipped "Clement Lefebvre <root@linuxmint.com>": No secret key
gpg: dpkg-sign.X9Cc9fCi/nemo_4.6.5.dsc: clear-sign failed: No secret key

dpkg-buildpackage: error: failed to sign .dsc file
```

Is this normal?

**Comment by icarter09** [comment]
@geajack yes that's normal. Use `sudo dpkg -i *deb` from your parent directory to install what you built.

**Comment by ItzSwirlz** [comment]
No no no. If you have the "no secret key" you need to enable the --no-sign flag to dpkg-buildpackage

**Comment by geajack** [comment]
What if I want to just build the application to test it or debug it? I don't want to have to install it with dpkg just to do that, I just want to run a binary. I noticed that when I ran `dpkg-buildpackage` a binary appeared in the repo at `./obj-x86_64-linux-gnu/src/nemo` which I was able to run, is there a way of easily just obtaining a binary like that without building the whole debian package and generating all those files in the parent directory?
